cockatrice/src/pictureloader.cpp

```diff
@@ -43,40 +43,80 @@ class PictureToLoad::SetDownloadPriorityComparator
     }
 };
 
-PictureToLoad::PictureToLoad(CardInfoPtr _card) : card(std::move(_card)), setIndex(0)
+PictureToLoad::PictureToLoad(CardInfoPtr _card) : card(std::move(_card))
 {
+    // This will be replaced with an expandable list ideally
+    urlTemplates.append(settingsCache->getPicUrl());
+    urlTemplates.append(settingsCache->getPicUrlFallback());
+
     if (card) {
         sortedSets = card->getSets();
         qSort(sortedSets.begin(), sortedSets.end(), SetDownloadPriorityComparator());
+        nextSet();
+    }
+}
+
+void PictureToLoad::populateSetUrls()
+{
+    currentSetUrls.clear();
+
+    if (card && currentSet) {
+        QString setCustomURL = card->getCustomPicURL(currentSet->getShortName());
+
+        if (!setCustomURL.isEmpty()) {
+            currentSetUrls.append(setCustomURL);
+        }
+    }
+
+    foreach (QString urlTemplate, urlTemplates) {
```

Comment:
The `urlTemplates` list contains a local copy of the unmodified picUrl strings from settingCache; I don't see it being useful in the present since we could just fetch the data from settingsCache here. Maybe you ahve an idea of using it to filter/modify the data coming from settingsCache?

Reply:
Yes, I am planning on updating the settingCache to contain a list of Urls instead of just two.  This is in line with what #2479 was proposing.  It will just make things easier to add in the future.

Reply:
My goal was to remove foreach from the codebase. Will get back to fixing this at some point in the future

Reply:
@ZeldaZach in favor of using `map` and `fold` functions? Nothing seems really terrible with foreach

Reply:
@ZeldaZach I'm newish to C++, what's the problem / prefered alternative for `foreach`?